languages/js/src/helpers.ts

```diff
@@ -306,6 +306,7 @@ let RESET = '';
 let FG_BLUE = '';
 let FG_RED = '';
 if (
+  typeof WorkerGlobalScope !== 'function' &&
```

Comment:
This check is meant to determine whether we are running in Node.

Instead of having an exception for `window` (browser) and `WorkerGlobalScope` (web workers), we could invert the check to check that the `process` object exists.

Reply:
@dhatch Thanks David!
Both WebWorkers and Node have access to a `process` object, but we could use the contents of the `title` property of that object.
https://nodejs.org/api/process.html#process_process_title

Making the change:
```js
if (process.title === 'node' &&
    typeof process.stdout.getColorDepth === 'function' &&
    process.stdout.getColorDepth() >= 4 &&
    typeof process.stderr.getColorDepth === 'function' &&
    process.stderr.getColorDepth() >= 4) {
    // ...
}
```
Do you see any foot-guns with that approach?

Reply:
Ah, didn't realize that.  My only concern would be if the process title is changed the first check may fail.

Instead, could we used the [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) operator to check that `process` `process.stdout` and `process.getColorDepth` exist?

Something like:

```
if (typeof process?.stdout?.getColorDepth === 'function' &&
    process.stdout.getColorDepth() >= 4 &&
    typeof process?.stderr?.getColorDepth === 'function' &&
    process.stderr.getColorDepth() >= 4) {
    // ...
}
```

What do you think?

Reply:
I very much like that. It's also consistent with the approach used elsewhere when webpack was being considered.
(for example: https://github.com/osohq/oso/commit/3f3528544bf0800ab7ec15d9917abfe4efaee0f1#diff-3dadef5973027043ecc1b5b8300cac77bbe0c54ecb26299bffec54e9224a82e7R1-R2)

I have pushed that suggestion. Thanks for talking that through!